### PR TITLE
CMDCT-3727: adding rate data from dynamo table to mathematica s3 bucket

### PR DESCRIPTION
**Pull request by angelaco11**
### Description
Uploading combined rate data (from dynamo rate table) to mathematica s3 bucket.

Also slightly not part of the scope of this ticket, but this dynamo sync functionality has been broken since qmr switched from AWS sdk v2 to v3. This PR includes Ben's changes to fix that (thank you @benmartin-coforma)

---
### How to test
1. Login to the mdct-qmr-dev AWS console, go to s3 and find the dynamosnapshot bucket for my branch (link [here](https://us-east-1.console.aws.amazon.com/s3/buckets/uploads-cmdct-3727-dynamosnapshots-024259748323?region=us-east-1&bucketType=general&tab=objects))
2. Make sure that along with `CSVmeasures` and `CSVcoreSet` you see a `CSVrate` folder in the bucket
3. Click into the `CSVrate` folder and make sure there is at least one <date>.csv file in there and that at least one timestamp is from within the last hour
4. Do steps 2 & 3 with the JSONrate folder

---
### Pre-review checklist

- [ ] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
- [ ] I have updated relevant documentation, if necessary
- [ ] I have performed a self-review of my code
- [ ] I have manually tested this PR in the deployed cloud environment

---
#### Review
- [ ] Design: This work has been reviewed and approved by design, if necessary
- [ ] Product: This work has been reviewed and approved by product owner, if necessary

#### Security
_If either of the following are true, notify the team's ISSO (Information System Security Officer)._

- [ ] These changes are significant enough to require an update to the SIA.
- [ ] These changes are significant enough to require a penetration test.
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

**Review by benmartin-coforma — APPROVED** [review]
As co-author, I approve 😎

File: services/uploads/src/dynamoSync/handler.js [hunk 1]
```diff
@@ -18,57 +18,44 @@ const uploadFileToS3 = async (filePath, scanResult) => {
...
-
 const syncDynamoToS3 = handler(async (_event, _context) => {
   console.log("Syncing Dynamo to Uploads");
-  const measureScanResults = await scanAll(process.env.measureTableName);
-  const coreSetScanResults = await scanAll(process.env.coreSetTableName);
-  const measureResults = measureScanResults ? measureScanResults : [];
-  const coreSetResults = coreSetScanResults ? coreSetScanResults : [];
+  const measureResults = await scanAll(process.env.measureTableName);
+  const coreSetResults = await scanAll(process.env.coreSetTableName);
+  const rateResults = await scanAll(process.env.rateTableName);
```
Comment (angelaco11) [inline services/uploads/src/dynamoSync/handler.js:32]: changes from 32 - 58 are the changes to add rate table to s3 bucket as part of cmdct-3727

File: services/uploads/src/libs/dynamodb-lib.js
Comment (angelaco11) [inline services/uploads/src/libs/dynamodb-lib.js:1]: Changes in this file were done by @benmartin-coforma to update the way we paginate through dynamo data (going from homegrown pagination to using aws sdk `paginateScan`)

File: services/uploads/src/dynamoSync/handler.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
 const { S3Client, PutObjectCommand } = require("@aws-sdk/client-s3");
 const { handler } = require("../libs/handler-lib");
-const { scan } = require("../libs/dynamodb-lib");
+const { scanAll } = require("../libs/dynamodb-lib");
```
Comment (angelaco11) [inline services/uploads/src/dynamoSync/handler.js:3]: changes from line 3 - 22 fix the issues that broke this upload to s3 lambda when updating from aws sdk v2 -> v3

**Review by ailZhou — APPROVED** [review]
LGTM! I see data being generated in the s3 bucket on your branch.

Thank you and @benmartin-coforma for fixing the upload issue and adding the rate data in as part of the data being sent!

**Pull request merged**